### PR TITLE
CRUD for smart alerts

### PR DESCRIPTION
**Pull request by swethalohith2**
## What

CRUD for smart alerts

- [x] GET
- [x] CREATE
- [x] DELETE
- [x] UPDATE

## References
[Kanban Card](https://instana.kanbanize.com/ctrl_board/132/cards/133967/details/)

### CODE REVIEW

**Comment by rongzhus** [comment]
@swethalohith2 Please add smart alert part to README file.

File: README.md [hunk 1]
```diff
@@ -654,4 +654,55 @@ synctl get app --name-filter <application-name>
...
+### synctl create alert Examples
+```
+# get synthetic test
+synctl get test 
+synctl create alert --name "Smart-alert" \
+       --alert_channel "$ALERT_CHANNEL" \
+       --test "$SYNTHETIC_TEST" \
+        --violation_count 2
+
+# or schedule multiple synthetic tests
```
Comment (rui1234) [inline README.md:690]: the comment is around synthetic tests

File: README.md [hunk 2]
```diff
@@ -654,4 +654,55 @@ synctl get app --name-filter <application-name>
...
+       --violation_count 2
+```
+# Delete Smart alert
+### synctl delete alert Syntax
+```
+synctl delete alert [id...] 
+```
+### synctl delete alert Examples
+```
+# delete a smart test
```
Comment (rui1234) [inline README.md:703]: delete a smart alert

File: README.md [hunk 3]
```diff
@@ -654,4 +654,55 @@ synctl get app --name-filter <application-name>
...
+--description, -d <string>          the description of Smart Alerts
+--severity <int>                    the severity of alert is either 5 (Warning), or 10 (Critical)
+--alert_channel <id>                alert_channel
+```
+### synctl create alert Examples
+```
+# get synthetic test
+synctl get test 
+synctl create alert --name "Smart-alert" \
+       --alert_channel "$ALERT_CHANNEL" \
```
Comment (rongzhus) [inline README.md:686]: How to get the `$ALERT_CHANNEL` ? can we list it via Rest API ?

Comment (swethalohith2) [reply README.md:686]: yes, we can use API to get all the alerting channels: [https://instana.github.io/openapi/#operation/getAlertingChannels](https://instana.github.io/openapi/#operation/getAlertingChannels)

File: synctl [hunk 4]
```diff
@@ -902,6 +925,102 @@ class CredentialConfiguration(Base):
...
+        """set alert channels"""
+        self.smart_alert_config["alertChannelIds"] = alert_channels
+
+    def get_json(self):
+        """return payload as json"""
+        if len(self.smart_alert_config["syntheticTestIds"]) == 0:
+            print("Error: no synthetic tests, set --test <test-id> at least a synthetic test")
+            sys.exit(-1)
+        if len(self.smart_alert_config["alertChannelIds"]) == 0:
+            print("Error: no alert channel, set --alert_channel <alert-channel-id> at least an alert channel ")
```
Comment (rongzhus) [inline synctl:1019]: typo should be `--alert-channel.

File: synctl [hunk 5]
```diff
@@ -1234,6 +1353,38 @@ class SyntheticTest(Base):
...
+        if create_res.status_code == 200:
+            # extracting data in json format
+            data = create_res.json()
+            alert_id = data["id"]
+            alert_name = data["name"]
+            print(f"smart alert \"{alert_name}\" created, id is \"{alert_id}\"")
+        elif create_res.status_code == 429:
+            self.exit_synctl(-1, "Too Many Requests")
+        else:
+            print('create test failed, status code:', create_res.status_code)
```
Comment (rongzhus) [inline synctl:1384]: create smart alert failed..

File: synctl [hunk 6]
```diff
@@ -1774,6 +2076,60 @@ class SyntheticTest(Base):
...
+            # print alert json
+            print(json.dumps(single_alert[0]))
+        elif show_details is True:
+            self.__print_a_alert_details(single_alert)
+
+    def __print_a_alert_details(self, single_alert):
+        a_single_alert = single_alert[0]
+        print(self.fill_space("Name".upper(), 30), "value".upper())
+        for key, value in a_single_alert.items():
+            if key == 'createdAt' or key == 'modifiedAt' or key == 'observedAt':
```
Comment (rongzhus) [inline synctl:2115]: update the key to `created` and `initialCreated`
image

**Review by rongzhus — CHANGES_REQUESTED** [review]
- show alert enable/disable status when get alert list.
- convert created/initialCreated time with `self.format_time` when show alert details.
- add option `--enable true/false` to enable and disable an alert for update command.

File: synctl [hunk 7]
```diff
@@ -1774,6 +2076,63 @@ class SyntheticTest(Base):
                         output_lists.append(t)
         print('total:', len(output_lists))
 
+
+    def print_synthetic_alerts(self, alerts):
+        id_length = 25
+        max_label_length = 50
+        severity_length = 10
+        enable_length = 1
```
Comment (rongzhus) [inline synctl:2084]: enable_length is not used yet, and 1 is not enough.

**Review by rongzhus — APPROVED** [review]

**Pull request merged**